algorithms/src/snark/varuna/ahp/verifier/verifier.rs

```diff
@@ -93,29 +92,33 @@ impl<TargetField: PrimeField, SM: SNARKMode> AHPForR1CS<TargetField, SM> {
         // Compute the domains and batch sizes.
         for (batch_size, (circuit_id, circuit_info)) in batch_sizes.values().zip(circuit_infos) {
             let constraint_domain_time = start_timer!(|| format!("Constructing constraint domain for {circuit_id}"));
-            let constraint_domain =
-                EvaluationDomain::new(circuit_info.num_constraints).ok_or(AHPError::PolyTooLarge)?;
+            let constraint_domain = EvaluationDomain::new(circuit_info.num_constraints)
+                .ok_or_else(|| anyhow::anyhow!("Polynomial degree is too large"))?;
             end_timer!(constraint_domain_time);
 
             let variable_domain_time = start_timer!(|| format!("Constructing constraint domain for {circuit_id}"));
-            let variable_domain =
-                EvaluationDomain::new(circuit_info.num_public_and_private_variables).ok_or(AHPError::PolyTooLarge)?;
+            let variable_domain = EvaluationDomain::new(circuit_info.num_public_and_private_variables)
+                .ok_or_else(|| anyhow::anyhow!("Polynomial degree is too large"))?;
             end_timer!(variable_domain_time);
 
             let non_zero_a_time = start_timer!(|| format!("Constructing non-zero-a domain for {circuit_id}"));
-            let non_zero_a_domain = EvaluationDomain::new(circuit_info.num_non_zero_a).ok_or(AHPError::PolyTooLarge)?;
+            let non_zero_a_domain = EvaluationDomain::new(circuit_info.num_non_zero_a)
+                .ok_or_else(|| anyhow::anyhow!("Polynomial degree is too large"))?;
```

Comment:
The original message is a bit misleading (and vague), since what's happening here is that the matrix has too many entries for it to be interpolated over one of the available domains. Also, and regardless of this, now that we're writing error messages directly it would be good if this case mentions the concrete matrix, which should be done in lines 111 and 116 too.
```suggestion
                .ok_or_else(|| anyhow::anyhow!("Matrix A has too many non-zero entries"))?;
```